xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Attachments.xml

```diff
@@ -0,0 +1,1144 @@
+#if ("$!{request.forceTestRights}" == "1")#template("xwikivars.vm")#end
+#template('attachment_macros.vm')
+## Get attachments
+#if ("$!wikimacro.parameters.tags" != '')
+  #set ($tags = $!wikimacro.parameters.tags.split(','))
+  #set ($pageReferenceSet = $collectiontool.getSet())
+  #foreach ($tag in $tags)
+    #set ($references = $xwiki.tag.getDocumentsWithTag("$tag"))
+    #set ($discard = $pageReferenceSet.addAll($references))
+  #end
+  #set ($attachments = [])
+  #foreach ($pageReference in $pageReferenceSet)
+    #set ($document = $xwiki.getDocument("$pageReference"))
+    #set ($documentAttachments = $document.attachmentList)
+    #set ($discard = $attachments.addAll($documentAttachments))
+  #end
+#else
+  #set ($attachments = $doc.attachmentList)
+  #if ("$!wikimacro.parameters.page" != '')
+    #set ($document = $xwiki.getDocument("$!wikimacro.parameters.page"))
+    #set ($attachments = $document.attachmentList)
+  #end
+#end
+## Filter attachments
+#set ($confluencePatterns = "$!wikimacro.parameters.patterns")
+#if ($confluencePatterns == '')
+  #set($attachmentsFiltered = $attachments)
+#else
+  #set ($patterns = $!confluencePatterns.split(','))
+  #set($attachmentsFiltered = [])
+  #foreach ($attachment in $attachments)
+    #foreach ($pattern in $patterns)
+      #set ($matches = $attachment.filename.matches("(?i)$!pattern"))
+      #if ($matches)
+        #set ($discard = $attachmentsFiltered.add($attachment))
+        #break
+      #end
+    #end
+  #end
+#end
+{{/velocity}}
+{{velocity}}
+## Sort attachments
+#set ($confluenceSortBy = "$!wikimacro.parameters.sortBy")
+#if ($confluenceSortBy == 'date')
+  #set ($sortBy = 'date')
+#elseif ($confluenceSortBy == 'size')
+  #set ($sortBy = 'filesize')
+#elseif ($confluenceSortBy == 'name')
+  #set ($sortBy = 'filename')
+#elseif ($confluenceSortBy == 'creation date')
+  #set ($sortBy = 'date')
+#else
+{{warning}}
+Warning: sortBy parameter must be one of 'date', 'size', or 'name'
+{{/warning}}
+#end
+#set ($confluenceSortOrder = "$!wikimacro.parameters.sortOrder")
+#set ($reverseOrderProperties = ['filesize', 'date'])
+#if ($confluenceSortOrder == 'ascending')
+  #if ($reverseOrderProperties.contains($sortBy))
+    #set ($sortOrder = 'desc')
+  #else
+    #set ($sortOrder = 'asc')
+  #end
+#elseif ($confluenceSortOrder == 'descending')
+  #if ($reverseOrderProperties.contains($sortBy))
+    #set ($sortOrder = 'asc')
+  #else
+    #set ($sortOrder = 'desc')
+  #end
+#else
+{{warning}}
+Warning: sortOrder parameter must be one of 'ascending' or 'descending'
+{{/warning}}
+#end
+#set ($attachmentsFiltered = $collectiontool.sort($attachmentsFiltered, "$sortBy:$sortOrder"))
+## Display attchments
+{{html clean="false"}}
+#if ($attachmentsFiltered.size() &gt; 0)
+  #displayAttachments($attachmentsFiltered)
+  #deleteAttachmentModal()
+#else
+  &lt;p class="noitems"&gt;$services.localization.render('core.viewers.attachments.noAttachments')&lt;/p&gt;
```

Comment:
Missing XML escaping for the translation value.

Reply:
Actually I took this from the `attachment_macros.vm`, so is there a reason why XML escaping is needed here and not in the vm, or is it missing too in the vm?

Reply:
We need to fix ``attachmentsinline.vm`` also. See https://jira.xwiki.org/browse/XWIKI-19749 .